src/main/java/bbd/jportal2/JPortal2Arguments.java

```diff
@@ -0,0 +1,149 @@
+
+    @Parameter(names = {"--template-location", "-tl"}, description = "Freemarker template location. Default is <current_working_directory>/jportal2_templates")
+    private List<String> templateLocations = Arrays.asList(Paths.get(System.getProperty("user.dir"), "jportal2_templates").toString());
+
+    @Parameter(names = {"--flag", "-F"}, description = "Flags to pass to the generator")
+    private List<String> flags = new ArrayList<>();
+
+    @Parameter(names = {"--help", "-h", "-?"}, help = true)
+    private boolean help;
+
+
+    public String getLogFileName() {
+        return logFileName;
+    }
+
+    public List<String> getInputDirs() {
+        return inputDirs;
+    }
+
+    public List<String> getInputFiles() {
+        return inputFiles;
+    }
+
+    public List<String> getBuiltinSIProcessors() {
+        return builtinSIProcessors;
+    }
+
+    public List<String> getTemplateSIProcessors() {
+        return templateSIProcessors;
+    }
+
+    public List<String> getBuiltinPostProcessors() {
+        return builtinPostProcessors;
+    }
+
+    public List<String> getTemplatePostProcessors() {
+        return templatePostProcessors;
+    }
+
+    public List<String> getTemplateLocations() {
+        return templateLocations;
+    }
+
+    public List<String> getFlags() {
+        return flags;
+    }
+
+    public boolean isHelp() {
+        return help;
+    }
+
+    public void setLogFileName(String logFileName) {
+        this.logFileName = logFileName;
+    }
+
+    public void setInputDirs(List<String> inputDirs) {
+        this.inputDirs = inputDirs;
+    }
+
+    public void setInputFiles(List<String> inputFiles) {
+        this.inputFiles = inputFiles;
+    }
+
+    public void setBuiltinSIProcessors(List<String> builtinSIProcessors) {
+        this.builtinSIProcessors = builtinSIProcessors;
+    }
+
+    public void setTemplateSIProcessors(List<String> templateSIProcessors) {
+        this.templateSIProcessors = templateSIProcessors;
+    }
+
+    public void setBuiltinPostProcessors(List<String> builtinPostProcessors) {
+        this.builtinPostProcessors = builtinPostProcessors;
+    }
+
+    public void setTemplatePostProcessors(List<String> templatePostProcessors) {
+        this.templatePostProcessors = templatePostProcessors;
+    }
+
+    public void setTemplateLocations(List<String> templateLocations) {
+        this.templateLocations = templateLocations;
+    }
+
+    public void setFlags(List<String> flags) {
+        this.flags = flags;
+    }
+
+    public void setHelp(boolean help) {
+        this.help = help;
+    }
+
+    @Override
+    public String toString() {
+        return "JPortal2Arguments{" +
+                "logFileName='" + logFileName + '\'' +
+                ", inputDirs=" + inputDirs +
+                ", inputFiles=" + inputFiles +
+                ", builtinSIProcessors=" + builtinSIProcessors +
+                ", templateSIProcessors=" + templateSIProcessors +
```

Comment:
Can we do this with reflection?

Reply:
We should be using Lombok to generate that. The JAR is about 1.1MB though but will allow us to get rid of all the boilerplate code: getters, setters, builders, constructors, equals and hash methods as well as toString.